### PR TITLE
Fix parsec server retry loop on postgresql connection loss

### LINKED ISSUE #8658: InterfaceError: cannot call Connection.remove_listener(): connection has been released back to the pool
Sentry Issue: [PARSEC3-SERVER-T](https://scille.sentry.io/issues/5963640691/?referrer=github_integration)

```
CancelledError: 
(3 additional frame(s) were not displayed)
...
  File "asyncio/tasks.py", line 665, in sleep
    return await future

CancelledError: 
(7 additional frame(s) were not displayed)
...
  File "asyncio/tasks.py", line 656, in sleep
    await __sleep0()
  File "asyncio/tasks.py", line 650, in __sleep0
    yield

InterfaceError: cannot call Connection.remove_listener(): connection has been released back to the pool
(6 additional frame(s) were not displayed)
...
  File "asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
  File "contextlib.py", line 231, in __aexit__
    await self.gen.athrow(value)
  File "contextlib.py", line 231, in __aexit__
    await self.gen.athrow(value)
  File "contextlib.py", line 231, in __aexit__
    await self.gen.athrow(value)

CancelledError: 
  File "asyncio/locks.py", line 212, in wait
    await fut

CancelledError: 
(12 additional frame(s) were not displayed)
...
  File "starlette/routing.py", line 75, in app
    await response(scope, receive, send)
  File "parsec/asgi/rpc.py", line 832, in __call__
    await self._run_session(scope, receive, send)
  File "parsec/asgi/rpc.py", line 883, in _run_session
    return await super().__call__(scope, receive, send)
  File "starlette/responses.py", line 258, in __call__
    async with anyio.create_task_group() as task_group:
  File "anyio/_backends/_asyncio.py", line 593, in __aexit__
    raise CancelledError

Exception in ASGI application

```

### PR DESCRIPTION
**Pull request by vxgmichel**
## Describe your changes

Fix #8658 

Related to #8656 

This PR has minimal changes to make the mechanism to retry after a postgresql connection loss work as expected.

There might be better ways to address the issue though.

## Checklist

Before you submit this pull request, please make sure to:

- [x] Keep changes in the pull request as small as possible
   
- [x] Ensure the commit history is sanitized
   
- [x] Give a meaningful title to your PR
- [x] Describe your changes
   
- [x] Link any related issue in the description
   
- [x] Link any dependent pull request in the description

If this PR adds a new feature or fixes a bug:

- [ ] I have added or updated relevant tests

If this PR affects the user:

- [ ] I have created a news fragment

### CODE REVIEW

File: server/parsec/cli/run.py [hunk 1]
```diff
@@ -455,6 +455,17 @@ def run_cmd(
                     app_config=app_config,
                 )
             )
+        # Ignore some noisy cancellation
+        # Note that this is no longer necessary with the latest anyio version (version 4.6)
```
Comment (mmmarcos) [inline server/parsec/cli/run.py:459]: I suppose there are breaking changes if we upgrade `anyio`? Have you considered/tried it?

**Review by FirelightFlagboy — APPROVED** [review]

File: server/parsec/cli/run.py (hunk 1 above)
Comment (mmmarcos) [reply server/parsec/cli/run.py:459]: There is already an issue for this: https://github.com/Scille/parsec-cloud/issues/7356
I'll link this comment.

**Review by mmmarcos — APPROVED** [review]

File: server/parsec/asgi/__init__.py [hunk 2]
```diff
@@ -106,10 +107,25 @@ async def serve_parsec_asgi_app(
...
+            await server.serve()
+        tg.cancel_scope.cancel()
+
+    async with anyio.create_task_group() as tg:
+        await tg.start(server_task)
+        try:
+            await anyio.sleep_forever()
+        finally:
+            # Use should_exit to shutdown the server gracefully
+            server.should_exit = True
```
[the remark this answers is not in the document]
Comment (mmmarcos) [reply server/parsec/asgi/__init__.py:131]: So we've created issue https://github.com/Scille/parsec-cloud/issues/8656 to discuss this with @touilleMan

File: server/parsec/cli/utils.py [hunk 3]
```diff
@@ -103,7 +103,6 @@ def cli_exception_handler(debug: bool) -> Iterator[bool]:
         yield debug
 
     except KeyboardInterrupt:
-        click.echo("bye ;-)")
```
Comment (AureliaDolo) [inline server/parsec/cli/utils.py:106]: So you remove this bye because it should be handled by the one line 470 in the run.py ?

**Review by AureliaDolo — APPROVED** [review]

File: server/parsec/cli/utils.py (hunk 3 above)
Comment (vxgmichel) [reply server/parsec/cli/utils.py:106]: Yes exactly

**Pull request merged**